18_regexp_lab/01.py

```diff
@@ -1,12 +1,26 @@
-"""
-Write a program that reads data
-from property files.
-Each line in the file can either be:
-    An empty line
-    A comment line (Start with #)
-    A property line (of the form key = value)
-
-Write a program that takes a property file name and key
-as command line arguments and prints the requested value
-"""
-
+_author_ = 'mariag'
+
+import re
+import sys
+
+
+
+file_key_value = sys.argv[1]
+input_key = sys.argv[2]
+kvDic = {}
+
+
+#load key|value to dictionary
+with open(file_key_value, 'r') as f_kv:
+    for line in f_kv:
+        m = re.search('^(\w+)\s*=\s*(\w+)' , line)
+        if m is not None:
+            key = m.group(1)
+            value = m.group(2)
+            kvDic[key] = value
+
+if kvDic.has_key(input_key):
+    print input_key," = ",
+    print kvDic.get(input_key)
+else:
+    print "key:'{}' doesn't exist".format(input_key)

```

Comment:
* What if the value of a key has spaces in it as in `name = Foo Bar`?
* `if m is not None:` is better written as `if m:`
* Instead of `if kvDic.has_key(input_key):` I'd write `if input_key in kvDic`